Libraries/config/Headers/config/Config.h

```diff
@@ -0,0 +1,39 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#ifndef __config_config_h
+#define __config_config_h
+
+#include <libutil/Filesystem.h>
+#include <plist/Object.h>
+
+namespace config {
+
+class Config {
+public:
+    Config(libutil::Filesystem const *filesystem, std::string const &defaultConfigPath = "/var/db/xcbuild_defaults.plist");
```

Comment:
Do you have any issue with just returning an empty Config option in the case of failure, or would you really like a ext::optional<Config>?